src/main/java/org/breedinginsight/services/TraitService.java

```diff
@@ -446,4 +448,21 @@ public Trait setTraitActiveStatus(UUID programId, UUID traitId, Boolean active,
     public boolean exists(UUID traitId) {
         return traitDAO.existsById(traitId);
     }
+
+    public List<String> getAllTraitTags(UUID programId) {
+
+        // Get all traits
+        List<Trait> traits = traitDAO.getTraitsFullByProgramId(programId);
+
+        // Parse out tags into unique list
+        Set<String> tags = new HashSet<>();
+        for (Trait trait: traits) {
+            if (trait.getTags() != null) {
+                tags.addAll(trait.getTags());
+            }
+        }
+
+        tags.add(FAVORITES_TAG);
```

Comment:
What is the purpose of this favorites tag?

Reply:
There should always be a favorites tag as an option. This sends the favorites tag back in the options of tags even if a trait isn't assigned that tag yet.